src/libgeds/TcpTransport.cpp

```diff
@@ -550,6 +552,7 @@ bool TcpPeer::processEndpointRecv(int sock) {
         (*recvQueue_stats)--;
         auto work = *it;
         ctx->va = work->va;
+        ctx->setPromise = false;
```

Comment:
can't we just check if there is a promise assigned or NULL? why do we have to bloat the rx context by another bool...?

Reply:
There is no way to check if a promise has already been set: https://en.cppreference.com/w/cpp/thread/promise

Reply:
I'm setting the promise to null.